src/test/java/org/jenkinsci/plugins/docker/workflow/RegistryEndpointStepTest.java

```diff
@@ -181,8 +204,73 @@ public void stepExecutionWithCredentialsAndQueueItemAuthenticator() throws Excep
 
         // Bob does not have Credentials.USE_ITEM permission and should not be able to use the credential.
         r.assertBuildStatus(Result.FAILURE, p2.scheduleBuild2(0));
+        });
     }
 
+    @Issue("JENKINS-75679")
+    @Test public void noFilePathPickle() throws Throwable {
+        assumeDocker();
+        try (var agent = new SSHAgentContainer()) {
+            agent.start();
+            rr.then(r -> {
+                agent.register("remote");
+                var registryCredentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, "registryCreds", null, "me", "s3cr3t");
+                CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), registryCredentials);
+                var p = r.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(
+                    """
+                    node('remote') {
+                      mockDockerLogin {
+                        withDockerRegistry(url: 'https://my-reg:1234', credentialsId: 'registryCreds') {
+                          semaphore 'wait'
+                        }
+                      }
+                    }
+                    """, true));
+                var b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("wait/1", b);
+            });
+            @SuppressWarnings("deprecation")
+            var verboten = FilePathPickle.class.getName();
+            assertThat(StringEscapeUtils.escapeJava(Files.readString(rr.getHome().toPath().resolve("jobs/p/builds/1/program.dat"), StandardCharsets.ISO_8859_1)), not(containsString(verboten)));
+            rr.then(r -> {
+                var b = r.jenkins.getItemByFullName("p", WorkflowJob.class).getBuildByNumber(1);
+                SemaphoreStep.success("wait/1", null);
+                r.assertBuildStatusSuccess(r.waitForCompletion(b));
+                r.assertLogContains("docker login -u me -p ******** https://my-reg:1234", b);
+                r.assertLogNotContains("s3cr3t", b);
+            });
+        }
+    }
+
+    private static final class SSHAgentContainer extends GenericContainer<SSHAgentContainer> {
```

Comment:
A handy utility (superseding the agent support in `docker-fixtures`) that we might want to publish somewhere. Maybe in JTH though that would then need an `optional` dep on `testcontainers` which may be undesirable.

Reply:
I am currently prototyping something.